service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java

```diff
@@ -24,95 +28,87 @@
 import java.util.Map;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderType.BLANK_ORDER;
+import static uk.gov.hmcts.reform.fpl.enums.GeneratedOrderType.CARE_ORDER;
 import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HER_HONOUR_JUDGE;
 import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createOrders;
 import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
 
 @ExtendWith(SpringExtension.class)
 class GeneratedOrderServiceTest {
     private static final String LOCAL_AUTHORITY_CODE = "example";
+    private static final String LOCAL_AUTHORITY_NAME = "Example Local Authority";
     private static final String COURT_NAME = "Example Court";
     private static final String COURT_EMAIL = "example@court.com";
-    private static final String CONFIG = String.format("%s=>%s:%s", LOCAL_AUTHORITY_CODE, COURT_NAME, COURT_EMAIL);
+    private static final String COURT_CONFIG = String.format("%s=>%s:%s", LOCAL_AUTHORITY_CODE, COURT_NAME,
+        COURT_EMAIL);
+    private static final String LA_NAME_CONFIG = String.format("%s=>%s", LOCAL_AUTHORITY_CODE, LOCAL_AUTHORITY_NAME);
     private static final LocalDateTime NOW = LocalDateTime.now();
 
     private final Time time = () -> NOW;
 
     private DateFormatterService dateFormatterService = new DateFormatterService();
-    private HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration = new HmctsCourtLookupConfiguration(CONFIG);
+    private HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration = new HmctsCourtLookupConfiguration(
+        COURT_CONFIG);
+    private LocalAuthorityNameLookupConfiguration localAuthorityNameLookupConfiguration =
+        new LocalAuthorityNameLookupConfiguration(LA_NAME_CONFIG);
 
     private GeneratedOrderService service;
 
     @BeforeEach
     void setup() {
-        this.service = new GeneratedOrderService(dateFormatterService, hmctsCourtLookupConfiguration, time);
+        this.service = new GeneratedOrderService(dateFormatterService, hmctsCourtLookupConfiguration,
+            localAuthorityNameLookupConfiguration, time);
     }
 
     @Test
-    void shouldAddDocumentToEmptyOrderWhenDocumentExists() throws IOException {
+    void shouldAddDocumentToOrderTypeAndDocumentObjectWhenDocumentExists() throws IOException {
```

Comment:
what about `shouldAddDocumentToOrderWhenDocumentExists`

Reply:
It doesn't strictly add it to the order until aboutToSubmit. This is in the mid event where the OrderTypeAndDocument object is updated with the order